### PR TITLE
schema: provide a jsonschema for dvc.yaml

### LINKED ISSUE #4027: Provide json schema for `dvc.yaml`
https://github.com/redhat-developer/vscode-yaml provides yaml validation and autocompletion as we type. It'd be a good DX to provide json-schema.

### PR DESCRIPTION
**Pull request by skshetry**
Fix #4027

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

If using VSCode, install `yaml by RedHat` and add the following config:
```json
"yaml.schemas": {
    "https://raw.githubusercontent.com/skshetry/dvc/dvc-yaml-jsonschema/scripts/schema/dvc-yaml.json":
      "dvc.yaml"
}
```

### CODE REVIEW

File: scripts/schema/dvc-yaml.json [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          ]
+        }
+      },
+      "additionalProperties": false
+    }
+  },
+  "required": [
+    "stages"
+  ]
+}
```
Comment (efiop) [inline scripts/schema/dvc-yaml.json:227]:
```suggestion
}

```

**Comment by efiop** [comment]
@skshetry What about docs, btw?

**Comment by efiop** [comment]
Not a VSCode user, so would really appreciate if anyone else could take a look. @Suor you are also using VSCode, right?

**Comment by skshetry** [comment]
@efiop, created https://github.com/iterative/dvc.org/issues/1429 and assigned myself to it. I'll work on it when 1.0 docs is ready.

**Comment by efiop** [comment]
@skshetry Added to the next sprint. Would be a shame to merge and not have the docs for it for a long time :slightly_smiling_face:

File: scripts/schema/dvc-yaml.json [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                  }
+                ]
+              },
+              "minItems": 1,
+              "uniqueItems": true
+            },
+            "frozen": {
+              "type": "boolean",
+              "description": "Assume stage as unchanged",
+              "default": true
```
Comment (skshetry) [inline scripts/schema/dvc-yaml.json:199]:
```suggestion
              "default": false
```

**Pull request merged**